### PR TITLE
feat(bloom-filter): add bloom filter reader

### PR DESCRIPTION
**Pull request by zhongzc**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)
#5176 

## What's changed and what's your intention?

Added `BloomFilterReader`:
1. Support prefetch when reading meta
2. `range_read` retained to facilitate cache support

## PR Checklist
Please convert it to a draft if some of the following conditions are not met.

- [x] I have written the necessary rustdoc comments.
- [x] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.
- [ ] API changes are backward compatible.
- [ ] Schema or data changes are backward compatible.

### CODE REVIEW

File: src/index/src/bloom_filter/reader.rs [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+use std::ops::Range;
+
+use async_trait::async_trait;
+use bytes::Bytes;
+use common_base::range_read::RangeReader;
+use fastbloom::BloomFilter;
+use snafu::{ensure, ResultExt};
+
+use super::error::{DeserializeJsonSnafu, IoSnafu};
+use super::{BloomFilterSegmentLocation, SEED};
```
Comment (waynexia) [inline src/index/src/bloom_filter/reader.rs:24]: nit: `super` path

File: src/index/src/bloom_filter/reader.rs [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+const BLOOM_META_LEN_SIZE: u64 = 4;
+
+/// Default prefetch size of bloom filter meta.
+pub const DEFAULT_PREFETCH_SIZE: u64 = 1024; // 1KiB
+
+/// `BloomFilterReader` reads the bloom filter from the file.
+#[async_trait]
+pub trait BloomFilterReader {
+    /// Reads range of bytes from the file.
+    async fn range_read(&mut self, offset: u64, size: u32) -> Result<Vec<u8>>;
```
Comment (waynexia) [inline src/index/src/bloom_filter/reader.rs:38]:
```suggestion
    async fn range_read(&mut self, offset: u64, size: u32) -> Result<Bytes>;
```

**Review by waynexia — APPROVED** [review]
Others LGTM

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**